src/lib/mgba/websocket-client.ts

```diff
@@ -13,440 +13,386 @@ export interface MgbaEvalResponse {
-    if (useCache) {
-      const cached = this.memoryCache.get(cacheKey)
-      if (cached && !cached.dirty && (now - cached.lastUpdated) < this.sharedBufferConfig.cacheTimeout) {
-        return cached.data
-      }
-    }
-
-    // Read from memory using optimized method
-    const data = await this.readBytes(address, size)
-
-    // Cache the result
-    if (useCache) {
-      this.memoryCache.set(cacheKey, {
-        address,
-        size,
-        data: new Uint8Array(data), // Make a copy
-        lastUpdated: now,
-        dirty: false,
-      })
-
-      // Cleanup cache periodically (every 10 accesses) to avoid overhead
-      this.cacheAccessCount++
-      if (this.cacheAccessCount % 10 === 0) {
-        this.cleanupCache()
-      }
-    }
+  isEvalConnected(): boolean {
+    return this.evalConnected
+  }
 
-    return data
+  /**
+   * Check watch connection status (for compatibility)
+   */
+  isWatchConnected(): boolean {
+    return this.watchConnected
   }
 
   /**
-   * Write data to shared buffer and mark cache as dirty
+   * Preload shared buffers (for compatibility)
    */
-  async setSharedBuffer (address: number, data: Uint8Array): Promise<void> {
-    // Write to memory
-    await this.writeBytes(address, data)
-
-    // Mark related cache entries as dirty
-    for (const [, region] of this.memoryCache.entries()) {
-      const regionEnd = region.address + region.size
-      const writeEnd = address + data.length
-
-      // Check if write overlaps with cached region
-      if (address < regionEnd && writeEnd > region.address) {
-        region.dirty = true
-      }
-    }
+  async preloadSharedBuffers(): Promise<void> {
+    // Simplified: start watching preload regions
+    await this.startWatchingPreloadRegions()
   }
 
   /**
-   * Invalidate specific cache entry
+   * Clear cache (for compatibility - now a no-op)
    */
-  invalidateCache (address: number, size: number): void {
-    const cacheKey = `${address}-${size}`
-    this.memoryCache.delete(cacheKey)
+  clearCache(): void {
+    // Simplified: no separate cache to clear
   }
 
   /**
-   * Clear all cached memory
+   * Invalidate cache (for compatibility - now a no-op)
    */
-  clearCache (): void {
-    this.memoryCache.clear()
+  invalidateCache(_address: number, _size: number): void {
+    // Simplified: no separate cache to invalidate
   }
 
   /**
-   * Clean up old cache entries
+   * Get cache stats (for compatibility)
    */
-  private cleanupCache (): void {
-    const now = Date.now()
-    const entries = Array.from(this.memoryCache.entries())
-
-    // Remove expired entries
-    for (const [key, region] of entries) {
-      if ((now - region.lastUpdated) > this.sharedBufferConfig.cacheTimeout) {
-        this.memoryCache.delete(key)
-      }
-    }
+  getCacheStats(): { hits: number, misses: number, size: number, regions: MemoryRegion[] } {
+    return { hits: 0, misses: 0, size: this.sharedBuffer.size, regions: this.watchedRegions }
+  }
```

Comment:
remove legacy compatibility code from this file. this should stand on its own and be a clean refactor, the other systems should adapt